controllers/glanceapi_controller.go

```diff
@@ -1320,6 +1330,36 @@ func (r *GlanceAPIReconciler) deleteJob(
 	return ctrlResult, err
 }
 
+// getEndpointID - returns the endpointID associated to a keystone Endpoint
+func (r *GlanceAPIReconciler) getEndpointID(
+	ctx context.Context,
+	instance *glancev1.GlanceAPI,
+) (string, error) {
+	ep := &keystonev1.KeystoneEndpoint{}
+	epID := ""
+	epType := endpoint.EndpointInternal
+	// in case of split, the external API will point to its own keystone Endpoint
+	// instead of looking for a different API that might not exist or have issues
+	// we can't control here
+	if instance.Spec.APIType == glancev1.APIExternal {
+		epType = endpoint.EndpointPublic
+	}
+	err := r.Get(ctx, types.NamespacedName{Namespace: instance.Namespace, Name: instance.Name}, ep)
+	if err != nil && !k8s_errors.IsNotFound(err) {
+		// Request object not found, GlanceAPI can't be executed with this config
+		r.Log.Info("EndpointID not ready")
+		instance.Status.Conditions.Set(condition.FalseCondition(
+			condition.ServiceConfigReadyCondition,
+			condition.ErrorReason,
+			condition.SeverityWarning,
+			condition.ServiceConfigReadyErrorMessage,
+			err.Error()))
+		return epID, nil
+	}
+	epID = ep.Status.EndpointIDs[string(epType)]
```

Comment:
If `err` was "not found", and therefore `ep` is still as it was post-L1338, is there a zero-value in `ep.Status.EndpointIDs` such that it won't hit a null pointer error?

Reply:
@abays I tried something like this [1] to see how golang deals with `omitempty` and fields that are not set in a struct (it would be `.Status` of the CR in our case), but if this changes (e.g., we remove `omitempty` in keystone), we might have a problem here, so I think we can do the assignment only when `EndpointIDs` is not `nil`.

[1] https://go.dev/play/p/Qd1HwxmuW6y